File: sample/test/test-echo.c
#include <stdio.h>
#include <unistd.h>

int main(int argc, char** argv)
{
	char c = '\0';

	printf("[Start test-1 example]\n");

	while (c != '\n') {
		if (read(STDIN_FILENO, &c, 1) != 1)
			return -1;

		putchar(c);
	}

	printf("[Success!]\n");
	return 0;
}

File: sample/test/test-echo.py
def harness(p):
    p.expect_exact("[Start test-1 example]")

    p.send("Hello world!")
    p.expect_exact("Hello world!")

    p.send("E")
    p.send("C")
    p.send("H")
    p.send("O")

    p.expect_exact("ECHO")

    p.sendline("")
    p.expect_exact("[Success!]")
